graphql/schema/schemagen.go

```diff
@@ -700,6 +723,63 @@ func genDgSchema(gqlSch *ast.Schema, definitions []string,
 		}
 	}
 
+	for fname, langTagDgPred := range langTagDgPreds {
+		dgPredAndTags := strings.Split(fname, "@")
+		unTaggedDgPredName := dgPredAndTags[0]
+		tags := dgPredAndTags[1]
+		gqlLangTaggedField := langTagDgPred.gqlField
+
+		if tags == "*" {
+			return "", gqlerror.ErrorPosf(gqlLangTaggedField.Position, "`*` language tag not"+
+				" supported in GraphQL, field: `%s`,type: `%s`", gqlLangTaggedField.Name,
+				langTagDgPred.gqlTypeName)
+		}
+		if unTaggedDgPred, ok := dgPreds[unTaggedDgPredName]; ok {
+			gqlUntaggedLangField := unTaggedDgPred.gqlField
+			if gqlUntaggedLangField.Type.Name() != "String" {
+				return "", gqlerror.ErrorPosf(gqlUntaggedLangField.Position, "Expected type:"+
+					" String, but got: `%s` for untagged language field `%s` inside type: `%s`",
+					gqlUntaggedLangField.Type.Name(), gqlUntaggedLangField.Name, unTaggedDgPred.gqlTypeName)
+			}
+			unTaggedDgPred.lang = true
+
+			if unTaggedDgPred.indexes["exact"] && langTagDgPred.indexes["hash"] {
+				return "", gqlerror.ErrorPosf(gqlUntaggedLangField.Position, "Incompatible indexes"+
+					" hash and exact are not allowed on language tagged and untagged fields,"+
+					" language untagged field: `%s` have exact index and language tagged field"+
+					" `%s` have hash index inside type: `%s`",
+					gqlUntaggedLangField.Name, gqlLangTaggedField.Name, unTaggedDgPred.gqlTypeName)
+			}
+
+			if unTaggedDgPred.indexes["hash"] && langTagDgPred.indexes["exact"] {
+				// if hash index is explicitly given on an untagged field and
+				// language tagged field have exact index then we return error for that case
+				search := gqlUntaggedLangField.Directives.ForName(searchDirective)
+				if search != nil {
+					indexes := search.Arguments.ForName(searchArgs)
+					if indexes != nil {
+						if x.HasString(getAllSearchIndexes(indexes.Value), "hash") {
+							return "", gqlerror.ErrorPosf(gqlUntaggedLangField.Position, "Incompatible indexes hash"+
+								" and exact are not allowed on language tagged and untagged fields,"+
+								" language untagged field: `%s` have hash index "+
+								"and language tagged field `%s` have exact index inside type: `%s`",
+								gqlUntaggedLangField.Name, gqlLangTaggedField.Name, unTaggedDgPred.gqlTypeName)
+						}
+					}
+				}
+				// if hash index is not explicitly given(i.e. @id directive applied) on an untagged field and
+				// language tagged field have exact index then we delete hash index from untagged
+				// language field because we add exact index on it
+				delete(unTaggedDgPred.indexes, "hash")
+			}
+
+			for index := range langTagDgPred.indexes {
+				unTaggedDgPred.indexes[index] = true
+			}
+			dgPreds[unTaggedDgPredName] = unTaggedDgPred
```

Comment:
Should we also be making one more validation ?
Checking if non-nullability of lang tag predicates and untagged predicates is same. Is it possible that name is of type String but name@hi is of type String!